modules/csp-overview.adoc

```diff
@@ -0,0 +1,35 @@
+// Module included in the following assemblies:
+//
+// * web_console/dynamic-plugin/content-security-policy.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="content-security-policy-overview_{context}"]
+= Content Security Policy (CSP) overview
+
+A Content Security Policy (CSP) is delivered to the browser in the `content-security-policy` response header. The policy is specified as a series of directives and values. Each directive type serves a different purpose, and each directive can have a list of values representing allowed sources.
+
+[id="content-security-policy-key-features_{context}"]
+== Key features of `ConsolePluginCSP`
+
+[discrete]
+=== Directive Types
+
+The supported directive types include `DefaultSrc`, `ScriptSrc`, `StyleSrc`, `ImgSrc`, and `FontSrc`. These directives allow you to specify valid sources for loading different types of content for your plugin. Each directive type serves a different purpose. For example, `ScriptSrc` defines valid JavaScript sources, while `ImgSrc` controls where images can be loaded from.
+
+//backporting the ConnectSrc directive, but that is tbd - openshift/console#14701
+
+[discrete]
+=== Values
+
+Each directive can have a list of values representing allowed sources. For example, `ScriptSrc` can specify multiple external scripts. These values are restricted to 1024 characters and cannot include whitespace, commas, or semicolons. Additionally, single-quoted strings and wildcard characters (`*`) are disallowed.
+
+[discrete]
+=== Unified Policy
+
+The {product-title} web console aggregates the CSP directives across all enabled `ConsolePlugin` custom resources (CRs) and merges them with its own default policy. The combined policy is then applied with the `Content-Security-Policy` HTTP response header.
```

Comment:
shall it be `Content-Security-Policy-Report-Only`header instead of `Content-Security-Policy` header? 
![Screenshot 2025-02-17 at 10 59 39 AM](https://github.com/user-attachments/assets/b1e6107c-91bf-472a-afb8-9ffd3e2cdf1b)

Reply:
Yes, we should. Updated the other to be `report-only` as well.